js/lib/electro_grammar_listener.js

```diff
@@ -0,0 +1,239 @@
+  exitTolerance(ctx) {
+    this.obj['tolerance'] = handle_unit(ctx, '');
+  }
+  exitMetric_size(ctx) {
+    var imperial_lookup = {
+      '0201': '008004',
+      '03015': '009005',
+      '0402': '01005',
+      '0603': '0201',
+      '1005': '0402',
+      '1608': '0603',
+      '2012': '0805',
+      '2520': '1008',
+      '3216': '1206',
+      '3225': '1210',
+      '4516': '1806',
+      '4532': '1812',
+      '4564': '1825',
+      '5025': '2010',
+      '6332': '2512',
+      '7451': '2920'
+    };
+    this.obj['package_size'] = imperial_lookup[handle_package_size(ctx)];
+  }
+  exitImperial_size(ctx) {
+    this.obj['package_size'] = handle_package_size(ctx);
+  }
+  exitAmbiguous_size(ctx) {
+    console.log('Warn: Ambiguous package size found');
+    this.obj['package_size'] = handle_package_size(ctx);
+  }
+  exitClass1(ctx) {
+    var dielectric;
+    if (ctx.M7G()) {
+      dielectric = 'M7G';
+    }
+    else if (ctx.C0G()) {
+      dielectric = 'C0G';
+    }
+    else if (ctx.H2G()) {
+      dielectric = 'H2G';
+    }
+    else if (ctx.L2G()) {
+      dielectric = 'L2G';
+    }
+    else if (ctx.P2H()) {
+      dielectric = 'P2H';
+    }
+    else if (ctx.R2H()) {
+      dielectric = 'R2H';
+    }
+    else if (ctx.S2H()) {
+      dielectric = 'S2H';
+    }
+    else if (ctx.T2H()) {
+      dielectric = 'T2H';
+    }
+    else if (ctx.U2J()) {
+      dielectric = 'U2J';
+    }
+    else if (ctx.Q3K()) {
+      dielectric = 'Q3K';
+    }
+    else if (ctx.P3K()) {
+      dielectric = 'P3K';
+    }
+
+    this.obj['dielectric'] = dielectric;
+  }
+  exitClass2(ctx) {
+    this.obj['dielectric'] = ctx.CLASS2().getText().toUpperCase();
+  }
+  exitAlu(ctx) {
+    this.obj['dielectric'] = 'ALU';
+  }
+  exitTan(ctx) {
+    this.obj['dielectric'] = 'TAN';
+  }
+  exitResistor(ctx) {
+    this.obj['type'] = 'resistor';
+  }
+  exitCapacitor(ctx) {
+    this.obj['type'] = 'capacitor';
+  }
+  exitInductor(ctx) {
+    this.obj['type'] = 'inductor';
+  }
+  exitOscillator(ctx) {
+    this.obj['type'] = 'oscillator';
+  }
+  exitRtype(ctx) {
+    if (ctx.POT())
+      this.obj['rtype'] = 'pot';
+  }
+  exitDiode(ctx) {
+    this.obj['type'] = 'diode';
+  }
+  exitDcode(ctx) {
+    this.obj['code'] = ctx.DCODE().getText().toUpperCase();
+  }
+  exitSignal(ctx) {
+    this.obj['dtype'] = 'signal';
+  }
+  exitRectifier(ctx) {
+    this.obj['dtype'] = 'rectifier';
+  }
+  exitLed(ctx) {
+    this.obj['dtype'] = 'led';
+  }
+  exitSchottky(ctx) {
+    this.obj['dtype'] = 'schottky';
+  }
+  exitZener(ctx) {
+    this.obj['dtype'] = 'zener';
+  }
+  exitColor(ctx) {
+    this.obj['color'] = ctx.COLOR().getText().toLowerCase();
+  }
+  exitTransistor(ctx) {
+    this.obj['type'] = 'transistor';
+  }
+  exitTtype(ctx) {
+    this.obj['ttype'] = ctx.TTYPE().getText().toLowerCase();
+  }
+  exitTcode(ctx) {
+    this.obj['code'] = ctx.TCODE().getText().toUpperCase();
+  }
+}
+
+exports.ElectroGrammarListener = ElectroGrammarToObjectListener;
```

Comment:
Naming it like this is confusing, I fixed this on lax-parser with https://github.com/monostable/electro-grammar/pull/8/commits/edfa7438e49f962780791e2be01436b3a78a5b0c.